rules.install

```diff
@@ -4,6 +4,45 @@
  * Rules - Installation file.
  */
 
+/**
+ * Implements hook_requirements().
+ */
+function rules_requirements($phase) {
+  $requirements = array();
+  if ($phase == 'update') {
+    // Required Backdrop modules that were not part of D7 installation.
+    $backdrop_modules = array(
+      'entity_ui' => 'Entity UI',
+      'entity_plus' => 'Entity Plus',
+    );
+    foreach ($backdrop_modules as $module => $module_name) {
+      if (module_exists($module)) {
+        // The module is already installed and enabled, nothing more to do.
+        continue;
+      }
+      $module_files = backdrop_system_listing('/' . $module . '\.module$/', 'modules', 'name', 0);;
```

Comment:
Not sure if this PR will be merge given the initiative to solve this in core. However, I wanted to point out that the preg pattern needs to have a "^" at the beginning to assure that you are getting the correct file. Without it, it may return things like `basic_entity_plus.module` and `entity_plus.module`. The "^" assures that the matches actually start with `$module`

`$module_files = backdrop_system_listing('/^' . $module . '\.module$/', 'modules', 'name', 0);

(Also you have two semi-colons 😄 )

Reply:
I'll fix the PR, but we should definitely hold off on merging to see what happens with https://github.com/backdrop/backdrop-issues/issues/5499.